costs/include/quadratic_cost.h

```diff
@@ -0,0 +1,85 @@
+#ifndef TORC_QUADRATIC_COST_H
+#define TORC_QUADRATIC_COST_H
+
+#include <string>
+#include "base_cost.h"
+
+namespace torc {
+    /**
+     * Class implementation of a linear cost function, f(x) = x^T A x, where A is a symmetric matrix.
+     * @tparam scalar_t the type of scalar used for the cost
+     */
+    template <class scalar_t>
+    class QuadraticCost: public BaseCost<scalar_t> {
+        using vectorx_t = Eigen::VectorX<scalar_t>;
+        using matrixx_t = Eigen::MatrixX<scalar_t>;
+
+    public:
+        /**
+         * Constructor for the quadratic cost class
+         * @param coefficients the A in x^T A x.
+         * @param identifier the name of the cost
+         */
+        QuadraticCost(const matrixx_t& coefficients,
+                      const std::string& identifier="Quadratic cost") {
+            if (coefficients.isUpperTriangular()) {
+                this->A_ = coefficients.template selfadjointView<Eigen::Upper>();
+            } else {
+                if ((coefficients.transpose() - coefficients).squaredNorm() != 0) {
+                    throw std::runtime_error("Quadratic cost: matrix must be either symmetric or upper triangular.");
+                }
+                this->A_ = coefficients;
+            }
+            this->identifier_ = identifier;
+            this->domain_dim_ = coefficients.cols();
+        }
+
+        /**
+         * Evaluates the cost function at a given point
+         * @param x the input to the function
+         * @return x^T A x
+         */
+        scalar_t Evaluate(const vectorx_t& x) const {
+            return x.dot(A_ * x);
+        }
+
+        /**
+         * Returns the A of the cost.
+         * @return the A_
+         */
+        matrixx_t GetCoefficients() const {
+            return A_;
+        }
+
+        /**
+         * Returns the gradient of the cost evaluated at x
+         * @param x the input
+         * @return grad f(x) = (A + A^T) x
+         */
+        vectorx_t Gradient(const vectorx_t& x) const {
+            return (A_ + A_.transpose()) * x;
```

Comment:
Since `A_` is symmetric we can just do `2*A_`. I think it will be more efficient.